pkg/virt-controller/watch/vmi.go

```diff
@@ -693,14 +690,18 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 			return nil
 		}
 	}
+	err := c.cleanupTriggerPods(pod, vmi)
```

Comment:
can we somehow converge the handling of these temp pods to a single place for cleanup? these code paths are really difficult to follow. that's not your fault, it's just the nature of how this packages has grown.

Reply:
I refactored the code to follow the same path in all cases.